modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

```diff
@@ -0,0 +1,77 @@
+:_content-type: PROCEDURE
+
+[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
+= Adding or removing pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance
+
+You can add or remove your choice of pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance that is cached by {prod-short} to support air-gapped, offline, and proxy-restricted environments.
+
+Completing this procedure results in an optional, custom build of the embedded Open VSX registry for use in workspaces of your organization’s Che.
+
+.Procedure
+
+include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]
+
+** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including an object with the extension `id` in the format __<published_by>__.__<unique_identifier>__ that you can collect from the extension page on link:https://open-vsx.org[open-vsx.org]:
++
+[source,json,subs="+quotes"]
+----
+    {
+        "id": "__<published_by>__.__<unique_identifier>__"
+    }
+----
++
+[TIP]
+====
+The latest extension version on link:https://open-vsx.org[open-vsx.org] is the default. Alternatively, you can add `"version": "__<extension_version>__"` on a new line to specify a version.
+====
+
+** If it is an extension from link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace] that is not available on link:https://open-vsx.org[Open VSX] yet, you can ask the extension author to publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions] so that you can add it as an extension that is published on link:https://open-vsx.org[open-vsx.org].
++
+[TIP]
+====
+If the extension author is not cooperative, you can add the extension directly from the `.vsix` file:
```

Comment:
This section should be removed. This implies "if you want something from the https://marketplace.visualstudio.com/VSCode and the author won't publish it to https://open-vsx.org, go ahead and just link to it."

This is 100% in violation of the TOU @ https://aka.ms/vsmarketplace-ToU

Reply:
I haven't read the TOU yet, but you can't link specifically to https://marketplace.visualstudio.com/VSCode because they hide the URL of the file.

Reply:
We can make it explicit that linking to https://marketplace.visualstudio.com/VSCode cannot and/or should not be attempted. Would that resolve your concern?

Reply:
OK. @nickboldt, I will remove this TIP then. I hope @svor is okay with this removal.
![image](https://user-images.githubusercontent.com/78564304/208548958-f984f931-593a-4d3e-bf04-415c58e99d48.png)

Reply:
I don't think we should remove this TIP. Maybe just make it as an another **if** in step 2,  because it could be useful when the extension (vsix file) is published on some custom openvsx registry, or on GitHub or on some web server.

Reply:
LGTM